working/2364 - primary constructors/feature-specification.md

```diff
@@ -246,20 +259,10 @@ is non-nullable (potentially non-nullable is enough).
 The current scope for the default values in the primary constructor is the
 enclosing library scope. This means that a naive copy/paste operation on
 the source code could change the meaning of the default value. In that case
-a new way to denote the given value is established. For example, consider
-this class using a primary constructor:
-
-```dart
-static const d = 42;
-
-class Point(int x, [int y = d]) {
-  void d() {}
-}
-```
-
-This corresponds to the following class without a primary constructor:
+a new way to denote the given value is established. For example, consider:
```

Comment:
This one is a little bit tricky because it is presenting a particular quirk of the desugaring process, and the presentation "primary then current" matches the desugaring process whereas "current then primary" uses the opposite ordering.

In particular, the "primary then current" ordering explains why we have that `_freshName` declaration (this is simply one of the things that the desugaring will do), but if we use a "current then primary" ordering then `_freshName` is just weird.

In all the other examples the ordering "current then primary" is fine, because they are focused on thinking about how to rewrite current code to use primary constructors.

Reply:
Updated to remove this as per our offline chat.